cabal.project: update cardano-base cardano-ledger-specs plutus deps

nfrisby opened the pull request:
Single commit, see its message.

JaredCorduan commented:
@nfrisby I fixed the problem in ledger (see https://github.com/input-output-hk/cardano-ledger-specs/pull/2525) and updated the ledger reference in this PR with the fix.

nfrisby commented:
> @nfrisby I fixed the problem in ledger (see [input-output-hk/cardano-ledger-specs#2525](https://github.com/input-output-hk/cardano-ledger-specs/pull/2525)) and updated the ledger reference in this PR with the fix.

Thank you, Jared! I confirm that Jared made just the one following change.

```
$ (cd ouroboros-network; git diff 2a9a879 23959f0)
diff --git a/cabal.project b/cabal.project
index 6340b6df1..d11b94311 100644
--- a/cabal.project
+++ b/cabal.project
@@ -212,8 +212,8 @@ source-repository-package
 source-repository-package
   type: git
   location: https://github.com/input-output-hk/cardano-ledger-specs
-  tag: 72d5b7de25b2271312082d48a5a081e564d9b79d
-  --sha256: 1bmq9ns3nyrgrgsg9b95cradx63yb1yx6387d4cv3iyl9gb9ji4r
+  tag: 16ace87553def9b422f70f921323539d45081edb
+  --sha256: 144f8cg6ar1d459k8nwnq85xjyrdfgly25h6msmkg2ra9rqj383n
   subdir:
     eras/alonzo/impl
     eras/alonzo/test-suite
```

And that that new commit is the latest in the `cardano-ledger-specs` repo.

```
$ (cd cardano-ledger-specs; git lg -3 origin/master)
16ace8755 - (origin/master, origin/HEAD) Merge pull request #2525 from input-output-hk/jc/fix-alonzo-txwit-serialization (2 hours ago) <Jared Corduan>
```

erikd commented:
bors merge
